includes/class-attachments.php

```diff
@@ -0,0 +1,665 @@
+<?php
+/**
+ * Attachments processing file.
+ *
+ * @package Activitypub
+ */
+
+namespace Activitypub;
+
+use Activitypub\Collection\Posts;
+
+/**
+ * Attachments processor class.
+ */
+class Attachments {
+	/**
+	 * Directory for storing ap_post media files.
+	 *
+	 * @var string
+	 */
+	public static $ap_posts_dir = '/activitypub/ap_posts/';
+
+	/**
+	 * Initialize the class and set up filters.
+	 */
+	public static function init() {
+		\add_action( 'before_delete_post', array( self::class, 'delete_attachments_with_post' ) );
+	}
+
+	/**
+	 * Delete attachments when an ap_post is deleted.
+	 *
+	 * @param int $post_id The post ID being deleted.
+	 */
+	public static function delete_attachments_with_post( $post_id ) {
+		if ( Posts::POST_TYPE !== \get_post_type( $post_id ) ) {
+			return;
+		}
+
+		self::delete_directory( $post_id );
+	}
+
+	/**
+	 * Delete the activitypub files directory for a post.
+	 *
+	 * @param int $post_id The post ID.
+	 */
+	public static function delete_directory( $post_id ) {
+		\WP_Filesystem();
+		global $wp_filesystem;
+
+		$activitypub_dir = \wp_upload_dir()['basedir'] . self::$ap_posts_dir . $post_id;
+
+		if ( $wp_filesystem->is_dir( $activitypub_dir ) ) {
+			$wp_filesystem->delete( $activitypub_dir, true );
+		}
+	}
+
+	/**
+	 * Import attachments from an ActivityPub object and attach them to a post.
+	 *
+	 * @param array $attachments Array of ActivityPub attachment objects.
+	 * @param int   $post_id     The post ID to attach files to.
+	 * @param int   $author_id   Optional. User ID to set as attachment author. Default 0.
+	 *
+	 * @return array Array of attachment IDs.
+	 */
+	public static function import( $attachments, $post_id, $author_id = 0 ) {
```

Comment:
```suggestion
	public static function import_for_post( $attachments, $post_id, $author_id = 0 ) {
```

to be a bit more specific and to follow the `delete_attachments_with_post` naming!?

Reply:
If `delete_attachments_with_post` is the inspiration, I think I'd rather change that to `delete_attachments`. "for post" or "with post" feel redundant

Reply:
Fixed in [0876ad0](https://github.com/Automattic/wordpress-activitypub/pull/2314/commits/0876ad0406f2df89a49e90c168419aaefa3da7c6). I ended up just re-using `delete_directory()`.

Reply:
That was not the only reason, I thought it might be nice to have all post specific functions prefixed because we want to (at least) also have an attachment importer for comments.

Reply:
Updated in [6820a9a](https://github.com/Automattic/wordpress-activitypub/pull/2314/commits/6820a9ab9b4afb9d37f8a82e86b37b47a1ccf523). It now has a public `import_post_files()` method in preparation for a future `import_comment_files()`. Under the hood they'll use the same private operations methods.